### PR TITLE
Wo/slack id to name

### PR DESCRIPTION
**Pull request by wogara**
created a slack channel names an id table. changed the task supervisor name and updating the slack controller to reflect that change.

### CODE REVIEW

File: server-phoenix/lib/helios/application.ex
Comment (chewnoill) [inline server-phoenix/lib/helios/application.ex:22]: nit: commit message is calling this a `rename` but it seems like its a new thing. Also not used yet?

File: server-phoenix/lib/helios/slack_channel_names.ex
Comment (bencates) [inline server-phoenix/lib/helios/slack_channel_names.ex:14]: Might make this easier to use to provide a default value for `query`

```suggestion
  def with_id(query \\ __MODULE__, channel_id) do
```

File: server-phoenix/lib/helios_web/controllers/web_hooks/slack_controller.ex
Comment (bencates) [inline server-phoenix/lib/helios_web/controllers/web_hooks/slack_controller.ex:21]: This line is a mouthful, both here and in `handle/2`. Might be worth pulling out a `defp slack_bearer_token()` to improve legibility.

File: server-phoenix/lib/helios_web/controllers/web_hooks/slack_controller.ex
Comment (bencates) [inline server-phoenix/lib/helios_web/controllers/web_hooks/slack_controller.ex:38]: This isn't really recovering from an error, it's just logging it and going on as though nothing went wrong. It's also spitting it out to stdout instead of properly logging it which may or may not get dropped in production depending on how the logs are captured and stored.

If we don't know how to programmatically recover from an error the best thing to do is probably to let it propagate up to the API layer and (hopefully) return a meaningful error message to the client. That way the user knows something broke and can react appropriately.

File: server-phoenix/lib/helios_web/controllers/web_hooks/slack_controller.ex
Comment (bencates) [inline server-phoenix/lib/helios_web/controllers/web_hooks/slack_controller.ex:64]: `response` is never used. We can just fire off `HTTPoison.post!/4`.

File: server-phoenix/lib/helios_web/controllers/web_hooks/slack_controller.ex
Comment (bencates) [inline server-phoenix/lib/helios_web/controllers/web_hooks/slack_controller.ex]: Again, not sure what we're rescuing here. If this isn't caught it'll propagate up to the `Task.Supervisor` which will log the crash for us.

**Review by bencates — COMMENTED** [review]
This looks pretty good! I added a few code cleanup suggestions.

**Comment by cathyzoller** [comment]
Is this PR a fork of your #417 branch?  If so, please note that in the description.  Doesn't look like this should be merged before #417, so perhaps also add a do not merge label until the first pr is merged.

**Comment by wogara** [comment]
> Is this PR a fork of your #417 branch? If so, please note that in the description. Doesn't look like this should be merged before #417, so perhaps also add a do not merge label until the first pr is merged.

no, the #417 PR is a fork of this branch

**Review by chewnoill — APPROVED** [review]
Nice work @wogara 🚀

**Pull request merged**
